### ISSUE TITLE
UriBuilder cannot build a correct mailto URI

### ISSUE THREAD

**Issue by jrdodds** [dotnet/runtime#87446]
### Description

The UriBuilder cannot be used to build a mailto URI.

### Reproduction Steps

The following code

```csharp
var builder = new UriBuilder
{
    Scheme = Uri.UriSchemeMailto,
    UserName = "someone",
    Host = "example.com",
    Query = "subject=Testing",
    Path = string.Empty,
};
Console.WriteLine(builder.ToString());
```

produces a URI value of

```
mailto:someone@example.com/?subject=Testing
```

which is not a valid `mailto` URI.

### Expected behavior

A valid URI would be

```
mailto:someone@example.com?subject=Testing
```

There should not be a '/' character.

### Actual behavior

The [`Path` setter](https://github.com/dotnet/runtime/blob/c32bee11a457151e5c400107ababed54dcd70438/src/libraries/System.Private.Uri/src/System/UriBuilder.cs#LL190C1-L202C1) sets the value to '/' when the value is the empty string.

        [AllowNull]
        public string Path
        {
            get => _path;
            set
            {
                _path = string.IsNullOrEmpty(value)
                    ? "/"
                    : Uri.InternalEscapeString(value.Replace('\\', '/'));
                _changed = true;
            }
        }

This makes sense for HTTP/S but makes the class unusable for mailto.

### Configuration

.Net 7.0.203
Windows 11 x64

**Comment by MihaZupan** [comment]
How did you run into this issue?
In an isolated example, the answer would be to just not set `Path` to empty, but you probably ran into it in more complex logic?

If this is causing issues, we probably could special-case this and avoid appending the slash, though I would also assume that this issue would be quickly spotted if someone did run into it.

**Comment by jrdodds** [comment]
> ... the answer would be to just not set `Path` to empty ...

Setting `Path` is something non-empty will also mangle the mailto URI.

e.g. setting a space:

```
mailto:someone@example.com ?subject=Testing
```

**Comment by MihaZupan** [comment]
Right, what I was getting at is that if you're constructing a mailto Uri like in your example, you can just skip setting `Path` - it will default to "actually empty" already.

**Comment by jrdodds** [comment]
> Right, what I was getting at is that if you're constructing a mailto Uri like in your example, you can just skip setting `Path` - it will default to "actually empty" already.

It doesn't behavior that way.

**Comment by mrudat** [comment]
Given that the mailto URI scheme has different functionality for formatting the URI, wouldn't the Right Way be to create a shiny new MailtoUri class that correctly formats a mailto URI and hooks into the existing UriBuilder and Uri?
